### PR TITLE
Update to JupyterLab 4.2 and Notebook 7.2

### PR DESCRIPTION
**Pull request by jtpio**
## References

Update to JupyterLab 4.2 and Notebook 7.2 pre-releases.

JupyterLab changelog: https://jupyterlab.readthedocs.io/en/latest/getting_started/changelog.html#v4-2

## Code changes

- [x] Update `@jupyterlab` packages
- [x] Update `@jupyter-notebook` packages
- [x] Update `@lumino` packages
- [x] Add new core plugins and extensions
- [x] Update migration guide

### Re-open recently closed files

image

### Dark High Contrast Theme

image

### Full notebook windowing mode by default

image

## Backwards-incompatible changes

Should be none.

### CODE REVIEW

**Comment by jtpio** [comment]
Looks like CI is now passing, and looking good on RTD (see screenshots above).

Will aim for a first `0.4.0` pre-release early next week.

File: packages/application/src/singleWidgetShell.ts [hunk 1]
```diff
@@ -85,9 +85,13 @@ export class SingleWidgetShell extends Widget implements JupyterFrontEnd.IShell
...
         return;
       }
+      const previousWidget = this.currentWidget;
       this._main.addWidget(widget);
       this._main.update();
-      this._currentChanged.emit(void 0);
+      this._currentChanged.emit({
+        newValue: widget,
+        oldValue: previousWidget,
+      });
```
Comment (Carreau) [inline packages/application/src/singleWidgetShell.ts:94]: 👍

File: ui-tests/test/kernels.spec.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ test.describe('Kernels', () => {
       .locator(`.jp-RunningSessions-item.jp-mod-kernel >> text="${name}"`)
       .waitFor();
     await page.locator('.jp-RunningSessions-item.jp-mod-kernel').first().hover();
-    await page.getByTitle('Shut Down Kernel').first().click();
+    await page.getByTitle('Shut Down').first().click();
```
Comment (Carreau) [inline ui-tests/test/kernels.spec.ts:69]: :+1:

**Review by Carreau — APPROVED** [review]
All looks good to me.

**Comment by jtpio** [comment]
Thanks @Carreau for the review!

**Pull request merged**